### PR TITLE
Restrain gc files by collection

### PR DESCRIPTION
**Pull request by congqixia**
Related to #21928

/kind improvement

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/datacoord/garbage_collector_test.go [hunk 1]
```diff
@@ -51,6 +53,132 @@ import (
...
+	})
+
+	s.Run("collectionPrefix_invalid", func() {
+		s.mockChunkManager.ExpectedCalls = nil
+		s.mockChunkManager.EXPECT().RootPath().Return("files")
+		s.mockChunkManager.EXPECT().ListWithPrefix(mock.Anything, mock.AnythingOfType("string"), mock.AnythingOfType("bool")).
+			Return([]string{"files/insert_log/1/", "files/bad_prefix", "files/insert_log/string/"}, lo.RepeatBy(3, func(_ int) time.Time {
+				return time.Now().Add(-time.Hour)
+			}), nil)
+		s.gc.option.collValidator = func(collID int64) bool {
```
Comment (yah01) [inline internal/datacoord/garbage_collector_test.go:134]: Need to mock files consist of both valid and invalid collections, and then check whether only the valid collections were removed

**Review by yah01 — CHANGES_REQUESTED** [review]

File: internal/datacoord/garbage_collector_test.go (hunk 1 above)
Comment (congqixia) [reply internal/datacoord/garbage_collector_test.go:149]: fixed

**Review by yah01 — APPROVED** [review]
/lgtm

**Pull request merged**